openhtf/util/data.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for utility functions that manipulate or compare data.
+
+We use a few special data formats internally, these utility functions make it a
+little easier to work with them.
+"""
+
+import difflib
+import logging
+import numbers
+
+from itertools import izip
+
+import mutablerecords
+
+from enum import Enum
+
+# Fields that are considered 'volatile' for record comparison.
+_VOLATILE_FIELDS = {'start_time_millis', 'end_time_millis', 'timestamp_millis',
+                    'sourcecode'}
+
+
+def AssertEqualsAndDiff(expected, actual):
+  """Compare two string blobs, log diff and raise if they don't match."""
+  if expected == actual:
+    return
+
+  # Output the diff first.
+  logging.error('***** Data mismatch:*****')
+  for line in difflib.unified_diff(
+      expected.splitlines(), actual.splitlines(),
+      fromfile='expected', tofile='actual', lineterm=''):
+    logging.error(line)
+  logging.error('^^^^^  Data diff  ^^^^^')
+
+  # Then raise the AssertionError as expected.
+  assert expected == actual
+
+
+def AssertRecordsEqualNonvolatile(first, second, indent=0):
```

Comment:
If this is meant to be a general-use method, I would pass in the `_VOLATILE_FIELDS` set here (and into the recursive calls)